packages/jupyter-widgets/src/manager/widgetLoader.ts

```diff
@@ -0,0 +1,89 @@
+import * as base from "@jupyter-widgets/base";
+import * as controls from "@jupyter-widgets/controls";
+
+let cdn = "https://unpkg.com";
+
+/**
+ * Constructs a well formed module URL for requireJS 
+ * mapping the modulename and version from the base CDN URL
+ * @param moduleName Name of the module corresponding to the widget package
+ * @param moduleVersion Module version returned from kernel
+ */
+function moduleNameToCDNUrl(moduleName: string, moduleVersion: string): string {
+  let packageName = moduleName;
+  let fileName = "index.js"; // default filename
+  // if a '/' is present, like 'foo/bar', packageName is changed to 'foo', and path to 'bar'
+  // We first find the first '/'
+  let index = moduleName.indexOf('/');
+  if (index !== -1 && moduleName[0] === '@') {
+    // if we have a namespace, it's a different story
+    // @foo/bar/baz should translate to @foo/bar and baz
+    // so we find the 2nd '/'
+    index = moduleName.indexOf('/', index + 1);
+  }
+  if (index !== -1) {
+    fileName = moduleName.substr(index + 1);
+    packageName = moduleName.substr(0, index);
+  }
+  let moduleNameString = packageName;
+  if(moduleVersion){
+    moduleNameString = `${packageName}@${moduleVersion}`;
+  }
+  return `${cdn}/${moduleNameString}/dist/${fileName}`;
+}
+
+/**
+ * Initialize dependencies that need to be preconfigured for requireJS module loading
+ * Here, we define the jupyter-base, controls package that most 3rd party widgets depend on
+ * We also override the cdn end point (if applicable)
+ */
+export function initRequireDeps(){
+  // Export the following for `requirejs`.
+  // tslint:disable-next-line: no-any no-function-expression no-empty
```

Comment:
I picked this up from the VS-Code implementation - https://github.com/microsoft/vscode-python/blob/329292c120490bdef69eaa6bc0997b4fd00beba8/src/ipywidgets/src/index.ts#L14. I believe, the tslint disable are an extra layer of caution in case people have very strict checks on disallowing function expressions or empty function declarations. `const define = () => {}` is the same as `const define = function () {}` and I think both these statements are function expressions because the function is assigned to a variable in an expression. So, I think the idea to disable the tslint rules makes sense to avoid TS compilation errors.

Reply:
`no-function-expression` means to use arrow functions instead of function(){}
![image](https://user-images.githubusercontent.com/1588988/98867874-2ac66600-2424-11eb-9a7d-489998890129.png)
https://developer.aliyun.com/mirror/npm/package/tslint-microsoft-contrib